week_3/Binary_Tree/class.js

```diff
@@ -0,0 +1,117 @@
+class Node {
+	constructor(val) {
+		this.val = val;
+		this.left = null;
+		this.right = null;
+	}
+}
+class BinarySearchTree {
+	constructor() {
+		this.root = null;
+	}
+	BFS() {
```

Comment:
what does `BFS` mean?

Reply:
BFS stands for Breadth First Search.

I should have been much more clear when I submitted this code. The class is something I wrote alongside the Data Structures and Algorithms Udemy course that I was talking to you about a few weeks ago.

Most of the methods introduced here are far beyond our needs. They involve traversal of a Binary Search Tree and, if needed for an algorithm at all, they would probably need to be rewritten.

Therefore, I'm going to pare down the class to its essentials: the constructor function and the "insert" method. The whole reason for this class is to quickly reproduce the correct data type for testing, and those two things will be sufficient for this purpose.

Reply:
Awesome 👍  sounds like a plan